garak/translator.py

```diff
@@ -0,0 +1,583 @@
+                                     ("authorization", "Bearer " + self.nim_api_key)])
+            self.nmt_client = riva.client.NeuralMachineTranslationClient(auth)
+
+    def _translate(self, text: str, source_lang: str, target_lang: str) -> str:
+        try:
+            if self.translation_service == "deepl":
+                return self.translator.translate_text(text, source_lang=source_lang, target_lang=target_lang).text
+            elif self.translation_service == "nim":
+                response = self.nmt_client.translate([text], "", source_lang, target_lang)
+                return response.translations[0].text
+        except Exception as e:
+            logging.error(f"Translation error: {str(e)}")
+            return text
+
+    def _get_response(self, input_text: str, source_lang: Optional[str] = None, target_lang: Optional[str] = None):
+
+        if not (source_lang and target_lang):
+            return input_text
+
+        translated_lines = []
+
+        for line in input_text.splitlines():
+            cleaned_line = remove_punctuation(line.strip().lower().split())
+            if cleaned_line.isspace() or cleaned_line.replace("-", "") == "":
+                translated_lines.append(cleaned_line)
+                continue
+            
+            translated_line = self._translate(cleaned_line, source_lang=source_lang, target_lang=target_lang)
+            translated_lines.append(translated_line)
+
+        res = '\n'.join(translated_lines)
+        logging.debug(f"translated_lines: {translated_lines}")
+
+        return res
+
+    def translate_prompts(self, prompts):
+        if hasattr(self, 'target_lang') is False or self.source_lang == "*" or self.target_lang == "":
+            return prompts 
+        translated_prompts = []
+        prompts = list(prompts)
+        for lang in self.target_lang.split(","):
+            if self.source_lang == lang:
+                continue 
+            for prompt in prompts:
+                mean_word_judge = is_english(prompt)
+                self.judge_list.append(mean_word_judge)
+                if mean_word_judge:
+                    translate_prompt = self._get_response(prompt, self.source_lang, lang)
+                    translated_prompts.append(translate_prompt)
+                else:
+                    translated_prompts.append(prompt)
+        if len(translated_prompts) > 0:
+            prompts.extend(translated_prompts)
+        return prompts 
+
+    def translate_triggers(self, triggers: list):
+        if is_nested_list(triggers):
+            trigger_list = []
+            for trigger in triggers:
+                trigger_words = self.translate_prompts(trigger)
+                for word in trigger_words:
+                    trigger_list.append([word])
+            triggers = trigger_list 
+            return triggers
+        else:
+            triggers = self.translate_prompts(triggers)
+            return triggers
+
+
+class DanTranslator(SimpleTranslator):
```

Comment:
Dan probe provide the long prompt.
If I translate it directly, it happens mistake translation.
I add some code for trying to avoid those case.

For example
- Long sentence split by "." or "?".
- Blank line
- Only exists "." or "?" or ". ".